src/Microsoft.AspNet.Mvc.Core/Formatters/MediaTypeComparisons.cs

```diff
@@ -0,0 +1,292 @@
+{
+    public static class MediaTypeComparisons
+    {
+        /// <summary>
+        /// Determines if the <paramref name="subset" /> media type is a subset of the <paramref name="set" /> media type
+        /// without taking into account the quality parameter.
+        /// </summary>
+        /// <param name="set">The more general media type.</param>
+        /// <param name="subset">The more specific media type.</param>
+        /// <returns><code>true</code> if <paramref name="set" />is a more general media type than <paramref name="subset"/>
+        /// except for the quality parameter; otherwise <code>false</code>.</returns>
+        public static bool IsSubsetOf(StringSegment set, string subset)
+        {
+            return IsSubsetOf(set, new StringSegment(subset), ignoreQuality: true);
+        }
+
+        /// <summary>
+        /// Determines if the <paramref name="subset" /> media type is a subset of the <paramref name="set" /> media type
+        /// without taking into account the quality parameter.
+        /// </summary>
+        /// <param name="set">The more general media type.</param>
+        /// <param name="subset">The more specific media type.</param>
+        /// <returns><code>true</code> if <paramref name="set" />is a more general media type than <paramref name="subset"/>
+        /// except for the quality parameter; otherwise <code>false</code>.</returns>
+        public static bool IsSubsetOf(string set, string subset)
+        {
+            return IsSubsetOf(new StringSegment(set), new StringSegment(subset), ignoreQuality: true);
+        }
+
+        /// <summary>
+        /// Determines if the <paramref name="subset" /> media type is a subset of the <paramref name="set" /> media type.
+        /// Two media types are compatible if one is a subset of the other ignoring any charset
+        /// parameter.
+        /// </summary>
+        /// <param name="set">The more general media type.</param>
+        /// <param name="subset">The more specific media type.</param>
+        /// <param name="ignoreQuality">Whether or not we should skip checking the quality parameter.</param>
+        /// <returns><code>true</code> if <paramref name="set" />is a more general media type than <paramref name="subset"/>;
+        /// otherwise <code>false</code>.</returns>
+        public static bool IsSubsetOf(StringSegment set, StringSegment subset, bool ignoreQuality)
+        {
+            if (!set.HasValue || !subset.HasValue)
+            {
+                return false;
+            }
+
+            try
+            {
+                var setMt = MediaTypeHeaderValue.Parse(set.Value);
+                var subSetMt = MediaTypeHeaderValue.Parse(subset.Value);
+
+                if (!ignoreQuality && setMt.Quality != subSetMt.Quality)
+                {
+                    return false;
+                }
+
+                return subSetMt.IsSubsetOf(setMt);
+            }
+            catch (Exception)
```

Comment:
now that we have only strings passing around an invalid media type can make it's way here, so in that case we just return false